core/camel-core-reifier/src/main/java/org/apache/camel/reifier/ThrottleReifier.java

```diff
@@ -33,30 +35,45 @@ public ThrottleReifier(Route route, ProcessorDefinition<?> definition) {
 
     @Override
     public Processor createProcessor() throws Exception {
+
         boolean async = parseBoolean(definition.getAsyncDelayed(), false);
         boolean shutdownThreadPool = willCreateNewThreadPool(definition, true);
         ScheduledExecutorService threadPool = getConfiguredScheduledExecutorService("Throttle", definition, true);
 
+        Expression correlation = null;
+        if (definition.getCorrelationExpression() != null) {
+            correlation = createExpression(definition.getCorrelationExpression());
+        }
+
+        boolean reject = parseBoolean(definition.getRejectExecution(), false);
         // max requests per period is mandatory
         Expression maxRequestsExpression = createMaxRequestsPerPeriodExpression();
         if (maxRequestsExpression == null) {
             throw new IllegalArgumentException("MaxRequestsPerPeriod expression must be provided on " + this);
         }
 
-        Expression correlation = null;
-        if (definition.getCorrelationExpression() != null) {
-            correlation = createExpression(definition.getCorrelationExpression());
-        }
+        if (ThrottlingMode.toMode(definition.getMode()) == ThrottlingMode.ConcurrentRequests) {
```

Comment:
Add parseString(definition...) as that does property placeholders

Reply:
I may need a clarification here. I'll ping you on Zulip / Slack.